File: leetcode/javascript/array/TwoPointers/checkIfPalindrome/checkIfPalindrome.js
/*
Учитывая строку s, верните true, если это палиндром, и false в противном случае.
Строка является палиндромом, если она читается как вперед, так и назад одинаково.
 */

/**
 * @param {string} s
 * @return {boolean}
 */
var checkIfPalindrome = function(s) {
    let left = 0;
    let right = s.length - 1;

    while (left < right) {
        if (s[left] !== s[right]) {
            return false;
        }

        left++;
        right--;
    }

    return true;
}

console.log(checkIfPalindrome("racecar"));
console.log(checkIfPalindrome("aleba"));

File: leetcode/go/array/TwoPointers/checkIfPalindrome(easy)/checkIfPalindrome.go
package main

import "fmt"

/*
Учитывая строку s, верните true, если это палиндром, и false в противном случае.
Строка является палиндромом, если она читается как вперед, так и назад одинаково.
*/

func main() {
	fmt.Println(checkIfPalindrome("racecar")) // true
	fmt.Println(checkIfPalindrome("aleba"))   // false
}

func checkIfPalindrome(s string) bool {
	left, right := 0, len(s)-1

	for left < right {
		if s[left] != s[right] {
			return false
		}
		left++
		right--
	}

	return true
}
